The headers generated by DXC from HLSL fail due to VS_INPUT not matching the VS_OUTPUT return type.

sys/d3d12/hlsl/VSMain_coord.h
"C:\Program Files (x86)\Windows Kits\10\bin\10.0.26100.0\\x64\dxc.EXE" "/Fh" "sys/d3d12/hlsl/VSMain_coord.h" "/E" "VSMain_coord" "/T" "vs_6_0" "/all-resources-bound" "../src/sys/d3d12/hlsl/VSMain_coord.hlsl"
../src/sys/d3d12/hlsl/VSMain_coord.hlsl:34:10: error: cannot initialize return object of type 'VS_OUTPUT' with an lvalue of type 'VS_INPUT'
  return input;

diff --git a/sys/d3d12/hlsl/VSMain_color.hlsl b/sys/d3d12/hlsl/VSMain_color.hlsl
--- a/sys/d3d12/hlsl/VSMain_color.hlsl
+++ b/sys/d3d12/hlsl/VSMain_color.hlsl
@@ -31,5 +31,8 @@
 
 VS_OUTPUT VSMain_color (VS_INPUT input)
 {
-  return input;
+  VS_OUTPUT output;
+  output.Position = input.Position;
+  output.Color = input.Color;
+  return output;
 }
diff --git a/sys/d3d12/hlsl/VSMain_coord.hlsl b/sys/d3d12/hlsl/VSMain_coord.hlsl
--- a/sys/d3d12/hlsl/VSMain_coord.hlsl
+++ b/sys/d3d12/hlsl/VSMain_coord.hlsl
@@ -31,5 +31,8 @@
 
 VS_OUTPUT VSMain_coord (VS_INPUT input)
 {
-  return input;
+  VS_OUTPUT output;
+  output.Position = input.Position;
+  output.Texture = input.Texture;
+  return output;
 }
diff --git a/sys/d3d12/hlsl/VSMain_pos.hlsl b/sys/d3d12/hlsl/VSMain_pos.hlsl
--- a/sys/d3d12/hlsl/VSMain_pos.hlsl
+++ b/sys/d3d12/hlsl/VSMain_pos.hlsl
@@ -29,5 +29,7 @@
 
 VS_OUTPUT VSMain_pos (VS_INPUT input)
 {
-  return input;
+  VS_OUTPUT output;
+  output.Position = input.Position;
+  return output;
 }
